From 05a2bbc5195533f9d06a780400aeec1bbab072fd Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Thu, 28 Aug 2014 14:23:56 -0400
Subject: [cpufreq] intel_pstate: Remove sample parameter in intel_pstate_calc_busy

Message-id: <1409235879-1283-143-git-send-email-prarit@redhat.com>
Patchwork-id: 89508
O-Subject: [RHEL7.1 PATCH BZ 1134639 142/185] intel_pstate: Remove sample parameter in intel_pstate_calc_busy
Bugzilla: 1134369
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Mark Langsdorf <mlangsdo@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1134639

commit 6b17ddb2a50b9403c6948ec3e4ea2bd2d7064ff3
Author: Stratos Karafotis <stratosk@semaphore.gr>
Date:   Tue Apr 29 20:53:49 2014 +0300

    intel_pstate: Remove sample parameter in intel_pstate_calc_busy

    Since commit d37e2b7644 ("intel_pstate: remove unneeded sample buffers")
    we use only one sample. So, there is no need to pass the sample
    pointer to intel_pstate_calc_busy. Instead, get the pointer from
    cpudata. Also, remove the unused SAMPLE_COUNT macro.

    While at it, reformat the first line in this function.

    Signed-off-by: Stratos Karafotis <stratosk@semaphore.gr>
    Acked-by: Viresh Kumar <viresh.kumar@linaro.org>
    Acked-by: Dirk Brandewie <dirk.j.brandewie@intel.com>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Cc: Lenny Szubowicz <lszubowi@redhat.com>
Cc: Tony Camuso <tcamuso@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/cpufreq/intel_pstate.c b/drivers/cpufreq/intel_pstate.c
index 1d777cf..e4c0985 100644
--- a/drivers/cpufreq/intel_pstate.c
+++ b/drivers/cpufreq/intel_pstate.c
@@ -32,8 +32,6 @@
 #include <asm/msr.h>
 #include <asm/cpu_device_id.h>
 
-#define SAMPLE_COUNT  3
-
 #define BYT_RATIOS  0x66a
 #define BYT_VIDS  0x66b
 #define BYT_TURBO_RATIOS 0x66c
@@ -553,14 +551,13 @@ static void intel_pstate_get_cpu_pstates(struct cpudata *cpu)
  intel_pstate_set_pstate(cpu, cpu->pstate.max_pstate);
 }
 
-static inline void intel_pstate_calc_busy(struct cpudata *cpu,
-     struct sample *sample)
+static inline void intel_pstate_calc_busy(struct cpudata *cpu)
 {
+ struct sample *sample = &cpu->sample;
  int32_t core_pct;
  int32_t c0_pct;
 
- core_pct = div_fp(int_tofp((sample->aperf)),
-   int_tofp((sample->mperf)));
+ core_pct = div_fp(int_tofp(sample->aperf), int_tofp(sample->mperf));
  core_pct = mul_fp(core_pct, int_tofp(100));
  FP_ROUNDUP(core_pct);
 
@@ -592,7 +589,7 @@ static inline void intel_pstate_sample(struct cpudata *cpu)
  cpu->sample.mperf -= cpu->prev_mperf;
  cpu->sample.tsc -= cpu->prev_tsc;
 
- intel_pstate_calc_busy(cpu, &cpu->sample);
+ intel_pstate_calc_busy(cpu);
 
  cpu->prev_aperf = aperf;
  cpu->prev_mperf = mperf;
-- 
1.7.1